src/cli/api/local_api.py

```diff
@@ -733,20 +713,189 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+        experiment_network_slug = experiment_data["asset"]["network"]["slug"]
+
+        # Check if the amount of nodes are valid for this network
+        experiment_nodes = experiment_data["asset"]["network"]["nodes"]
+
+        all_network_nodes = self._get_network_nodes()
+        network_nodes = all_network_nodes[experiment_network_slug]
+        if len(experiment_nodes) > len(network_nodes):
+            error_dict["error"].append(f"In file '{experiment_file_path}': too many nodes used in network "
+                                       f"'{experiment_network_slug}'. Maximum amount of nodes that can be used: "
+                                       f"{len(network_nodes)}")
+
+        # Check if all nodes exist and belong to this network
+        for node in experiment_nodes:
+            if node["slug"] not in network_nodes:
+                error_dict["error"].append(f"In file '{experiment_file_path}': '{node['slug']}' does not exist or does "
+                                           f"not belong to the network '{experiment_network_slug}'")
+
+    def _validate_experiment_channels(self, experiment_file_path: Path, experiment_data: Dict[str, Any], error_dict:
+                                     ErrorDictType) -> None:
+        """
+        Validate if the amount of channels (defined in the experiment.json file) are valid for 'network_slug' and if
+        all the channels exist and belong to this network.
+
+        Args:
+            experiment_file_path: The location of the experiment.json file
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+
+        # Check if the amount of channels are valid for this network
+        experiment_network_slug = experiment_data["asset"]["network"]["slug"]
+        experiment_channels = experiment_data["asset"]["network"]["channels"]
+        network_channels = self._get_channels_for_network(network_slug=experiment_network_slug)
+        if network_channels is not None:
+            if len(experiment_channels) > len(network_channels):
+                error_dict["error"].append(f"In file {experiment_file_path}: too many channels used in network "
+                                           f"'{experiment_network_slug}'. Maximum amount of channels that can be used: "
+                                           f"{len(network_channels)}")
+
+            # Check if the channels exist and belong to this network
+            for channel in experiment_channels:
+                if not channel["slug"] in network_channels:
+                    error_dict["error"].append(f"In file '{experiment_file_path}': '{channel['slug']}' does not exist "
```

Comment:
Channel {channel['slug']}' does not exist
This would make the error message more clear about what does not exist.

Reply:
Will add it